include/swift/SIL/SILInstruction.h

```diff
@@ -87,6 +88,125 @@ SILInstructionKind getSILInstructionKind(StringRef InstName);
 /// Map SILInstructionKind to a corresponding SILInstruction name.
 StringRef getSILInstructionName(SILInstructionKind Kind);
 
+/// A formal SIL reference to a list of values, suitable for use as the result
+/// of a SILInstruction.
+///
+/// *NOTE* Most multiple value instructions will not have many results, so if we
+/// want we can cache up to 3 bytes in the lower bits of the value.
+///
+/// *NOTE* Most of this defined out of line further down in the file to work
+/// around forward declaration issues.
+class SILInstructionResultArray {
+  const uint8_t *Pointer;
+  unsigned Size;
+
+public:
+  SILInstructionResultArray() : Pointer(nullptr), Size(0) {}
+  SILInstructionResultArray(const SingleValueInstruction *SVI);
+
+  SILInstructionResultArray(const SILInstructionResultArray &Other)
+      : Pointer(Other.Pointer), Size(Other.Size) {}
+
+  SILInstructionResultArray &operator=(const SILInstructionResultArray &Other) {
+    Pointer = Other.Pointer;
+    Size = Other.Size;
+    return *this;
+  }
+
+  SILInstructionResultArray(SILInstructionResultArray &&Other)
+      : Pointer(Other.Pointer), Size(Other.Size) {}
+
+  SILInstructionResultArray &operator=(SILInstructionResultArray &&Other) {
+    Pointer = Other.Pointer;
+    Size = Other.Size;
+    return *this;
+  }
```

Comment:
All four of these are exactly the default implementation; please remove them.

Reply:
Will do.